zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java

```diff
@@ -114,50 +114,57 @@ public void copyVersion(@Nonnull String projectSlug,
-                        + newVersionSlug);
+            projectSlug + ":" + versionSlug, projectSlug + ":"
+                + newVersionSlug);
 
         // Copy of HProjectIteration
         HProjectIteration newVersion =
-                projectIterationDAO.getBySlug(projectSlug, newVersionSlug);
+            projectIterationDAO.getBySlug(projectSlug, newVersionSlug);
 
-        newVersion = copyVersionSettings(version, newVersion);
-        newVersion = projectIterationDAO.makePersistent(newVersion);
+        try {
+            newVersion = copyVersionSettings(version, newVersion);
+            newVersion = projectIterationDAO.makePersistent(newVersion);
 
-        // Copy of HDocument
-        int docSize =
+            // Copy of HDocument
+            int docSize =
                 documentDAO.getDocCountByVersion(projectSlug, versionSlug);
 
-        int docStart = 0;
-        while (docStart < docSize) {
-            Map<Long, Long> docMap =
+            int docStart = 0;
+            while (docStart < docSize) {
+                Map<Long, Long> docMap =
                     copyDocumentBatch(version.getId(), newVersion.getId(),
-                            docStart, DOC_BATCH_SIZE);
-            docStart += DOC_BATCH_SIZE;
+                        docStart, DOC_BATCH_SIZE);
+                docStart += DOC_BATCH_SIZE;
 
-            for (Map.Entry<Long, Long> entry : docMap.entrySet()) {
-                // Copy of HTextFlow and HTextFlowTarget
-                copyTextFlowAndTarget(entry.getKey(), entry.getValue());
+                for (Map.Entry<Long, Long> entry : docMap.entrySet()) {
+                    // Copy of HTextFlow and HTextFlowTarget
+                    copyTextFlowAndTarget(entry.getKey(), entry.getValue());
 
-                if (taskHandleOpt.isPresent()) {
-                    taskHandleOpt.get().incrementDocumentProcessed();
-                    taskHandleOpt.get().increaseProgress(1);
+                    if (taskHandleOpt.isPresent()) {
+                        taskHandleOpt.get().incrementDocumentProcessed();
+                        taskHandleOpt.get().increaseProgress(1);
+                    }
                 }
             }
-        }
-
-        // restore version.status after complete
-        newVersion = projectIterationDAO.getBySlug(projectSlug, newVersionSlug);
-        newVersion.setStatus(version.getStatus());
-        projectIterationDAO.makePersistent(newVersion);
-        projectIterationDAO.flush();
-
-        // clear any cache that has been loaded in this new version before copy
-        // completed
-        versionStateCacheImpl.clearVersionStatsCache(newVersion.getId());
-        log.info("copy version end: copy {} to {}, {}", projectSlug
-                + ":" + versionSlug, projectSlug + ":" + newVersionSlug,
+        } catch (Exception e) {
+            log.error(
+                "Error during copy version from project '{}' {} to {}: {}",
```

Comment:
If you leave out the final `{}`, slf4j should log the exception stack trace with the log entry.